tests/test_codegen_directive.cpp

```
@@ -470,6 +542,76 @@ TEST_F(DirectiveTest, NativeFnLibraryDoesNotAffectDispatch) {
+        "@native(\"mylib\")\n"
+        "function greet(name: str) -> str\n"
+        "function greet(x: int) -> str:\n"
+        "    return \"int:\" + to_str(x)\n"
+        "print(greet(42))\n"
+    );
+    EXPECT_EQ(output, "int:42\n");
+}
+
+// ===== @native("libname") generic dispatch tests =====
+
+TEST_F(DirectiveTest, NativeFnLibraryGenericDispatchDirect) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)